addon-test-support/@ember/test-helpers/test-metadata.ts

```diff
@@ -0,0 +1,51 @@
+import { BaseContext } from './setup-context';
+
+export interface ITestMetadata {
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+  [key: string]: any;
+
+  readonly isRendering: boolean;
+  readonly isApplication: boolean;
+}
+
+export class TestMetadata implements ITestMetadata {
+  [key: string]: any;
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+
+  constructor() {
+    this.setupTypes = [];
```

Comment:
Other than application and rendering tests do we consider others as unit tests or container tests?

Reply:
So here are the classifications:
- Plain QUnit tests that don't use any features from `@ember/test-helpers`, including what we'd consider `unit` right now (`setupTypes` and `usedHelpers` are both completely empty)
- What is termed 'container' tests, which is effectively all tests built using `setupContext` within `@ember/test-helpers` (`setupTypes` does not contain `setupRenderingContext` and `setupApplicationContext`)
- rendering (covered already)
- application (covered already)

We only want to be explicit about `rendering` and `application`, as we have a plan to simplify the terminology of these tests types down the road.